### PR TITLE
Add a dialog for an nbformat conversion

### LINKED ISSUE #1426: Handle nbformat version compatibility with dialogs
cf https://github.com/jupyter/notebook/blob/50ec1326153cc86ea1b200e84e2dc57c265813f8/notebook/static/notebook/js/notebook.js#L2948

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #1426

**Screenshots**
image

### CODE REVIEW

**Comment by blink1073** [comment]
I verified that the dialog goes away when saving.

File: packages/notebook/src/model.ts [hunk 1]
```diff
@@ -243,6 +245,42 @@ export class NotebookModel extends DocumentModel implements INotebookModel {
...
+    // Alert the user if the format changes.
+    if (origNbformat !== undefined && this._nbformat !== origNbformat) {
+      let src;
+      if (this._nbformat > origNbformat) {
+        src = ' an older notebook format ';
+      } else {
+        src = ' a newer notebook format ';
+      }
+
+      let msg =
```
Comment (jasongrout) [inline packages/notebook/src/model.ts]: It would be much cleaner to use a backtick string here to interpolate variables.

Comment (blink1073) [reply packages/notebook/src/model.ts]: Modernized now :wink:

**Comment by jasongrout** [comment]
I was thinking of going a bit farther with the modernization :)

```typescript
      let newer = this._nbformat > origNbformat;
      let msg = `This notebook has been converted from ${
        newer ? 'an older' : 'a newer'
      } notebook format (v${origNbformat}) to the current notebook format (v${
        this._nbformat
      }). The next time you save this notebook, the current notebook format (v${
        this._nbformat
      }) will be used. ${
        newer
          ? ' Older versions of Jupyter may not be able to read the new format.'
          : ' Some features of the original notebook may not be available.'
      }  To preserve the original format version, close the notebook without saving it.`;
```

**Comment by jasongrout** [comment]
I pushed a change using template strings even more. Let me know if it looks good to you.

**Comment by blink1073** [comment]
LGTM.

**Comment by jasongrout** [comment]
I tested this using a v3 notebook, and it works great! I'm trying to figure out why the tests are not passing, though.

**Comment by blink1073** [comment]
My suspicion is that there may be an unaccepted dialog, but that would only happen if our notebook had a different nbformat version, either the external file that we load or inline data.

**Comment by jasongrout** [comment]
Yes, just tracked down this in the test: https://github.com/jupyterlab/jupyterlab/blob/cf1063590f3f438b771b679fafc2eeed0838421e/testutils/src/notebook-utils.ts#L78

**Comment by jasongrout** [comment]
Thanks for adding a test!

**Pull request merged**